File: backend/src/main/java/com/pepper/backend/services/messaging/MessageParserService.java
package com.pepper.backend.services.messaging;

import com.pepper.backend.model.messaging.Message;
import com.pepper.backend.model.messaging.Person;
import com.pepper.backend.model.messaging.Sender;
import com.pepper.backend.model.messaging.Task;
import org.springframework.stereotype.Service;

@Service
public class MessageParserService {

    public Message parse(String message) throws IndexOutOfBoundsException {
        String[] messageSplit = message.split("#", 2);

        String path = messageSplit[0];
        String data = messageSplit[1];

        String[] pathSplit = path.split(":");

        return Message.builder()
                .sender(Sender.valueOf(pathSplit[0]))
                .senderId(pathSplit[1])
                .person(Person.valueOf(pathSplit[2]))
                .personId(pathSplit[3])
                .task(Task.valueOf(pathSplit[4]))
                .taskId(pathSplit[5])
                .data(data.substring(1, data.length() - 1))
                .build();
    }

    public String stringify(Sender sender, String senderId, Person person, String personId, Task task, String taskId, String data) {
        return sender.toString() + ":"
                + senderId + ":"
                + person.toString() + ":"
                + personId + ":"
                + task.toString() + ":"
                + taskId + "#{"
                + data + "}";
    }

}

File: zorgmedewerker-angular/src/app/message-parser.service.ts
import { Injectable } from '@angular/core';
import { Message, Sender, Person, Task } from '../model/message';

@Injectable({
  providedIn: 'root'
})
export class MessageParserService {

  constructor() { }

  parse(message: string): Message {
    let temp = {} as Message;

    const messageSplit: string[] = message.split('#');

    const path: string = messageSplit[0];
    const data: string = messageSplit[1];

    const pathSplit: string[] = path.split(':');

    temp.sender = pathSplit[0] as unknown as Sender;
    temp.senderId = pathSplit[1];
    temp.person = pathSplit[2] as unknown as Person;
    temp.personId = pathSplit[3];
    temp.task = pathSplit[4] as unknown as Task;
    temp.taskId = pathSplit[5];
    temp.data = data.substring(1, data.length - 1);

    return temp;
  }

  stringify(sender: Sender, senderId: string, person: Person, personId: string, task: Task, taskId: string, data: string): string {
    return Sender[sender] + ':'
      + senderId + ':'
      + Person[person] + ':'
      + personId + ':'
      + Task[task] + ':'
      + taskId + '#{'
      + data + '}';
  }

}
